webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -427,16 +426,25 @@ const AdvancedTextEditor = ({
         };
     }, [handleDraftChange, draft]);
 
-    // Set the draft from store when changing post or channels, and store the previus one
+    // Keep track of the draft as a ref so that we can save it when changing channels
+    const draftRef = useRef(draftFromStore);
     useEffect(() => {
-        setDraft(draftFromStore);
-        return () => handleDraftChange(previousDraft.current, {instant: true, show: true});
-    }, [channelId, postId]);
+        draftRef.current = draft;
+    }, [draft]);
 
-    // Keep track of the previous draft
+    // Set the draft from store when changing post or channels, and store the previous one
     useEffect(() => {
-        previousDraft.current = draft;
-    }, [draft]);
+        // Store the draft that existed when we opened the channel to know if it should be saved
+        const draftOnOpen = draftFromStore;
+
+        setDraft(draftOnOpen);
+
+        return () => {
+            if (draftOnOpen !== draftRef.current) {
+                handleDraftChange(draftRef.current, {instant: true, show: true});
```

Comment:
It's intentional that this may be saving a draft that was already saved by the timeout in `handleDraftChange` previously. We need to save it again here to set `show` so that it shows in the LHS